[CMake][GTK] Introduce CMake Presets

obyknovenius opened the pull request:
#### b41931e0f27eff8e8ec100518f53bf9bc2970ed8
<pre>
[CMake][GTK] Introduce CMake Presets
<a href="https://bugs.webkit.org/show_bug.cgi?id=253304">https://bugs.webkit.org/show_bug.cgi?id=253304</a>

Reviewed by Adrian Perez de Castro.

CMake Presets allow easily share common configure and build options.
They can be used with command line cmake and parsed by IDEs,
such as Visual Studio Code or JetBrains CLion,
to automatically configure the project and setup various integrations.

Instead of
```
mkdir WebKitBuild/Release
cd WebKitBuild/Release
cmake -DPORT=gtk -DDEVELOPER_MODE=ON -DCMAKE_BUILD_TYPE=Release ../..
cmake --build .
```
it&apos;s now possible to do just
```
cmake --preset=gtk-release
cmake --build --preset=gtk-release
```

* CMakePresets.json: Added.

Canonical link: <a href="https://commits.webkit.org/261422@main">https://commits.webkit.org/261422@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/631ab9af17aefb57c2728103315338e9663e1ee0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111682 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20818 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/295 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3437 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/120414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22173 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11898 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/3171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117448 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16465 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99615 "Build is being retried. Recent messages:OS: Monterey (12.6.3), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; 4 api tests failed or timed out") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/104612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/104612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13297 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/89/builds/192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/104612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13787 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/9635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19232 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52189 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15777 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4336 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
I would be neat to have WPE and JSCOnly presets as well, but those can be added in a follow-up patch as well.

obyknovenius committed:
[CMake][GTK] Introduce CMake Presets
https://bugs.webkit.org/show_bug.cgi?id=253304

Reviewed by Adrian Perez de Castro.

CMake Presets allow easily share common configure and build options.
They can be used with command line cmake and parsed by IDEs,
such as Visual Studio Code or JetBrains CLion,
to automatically configure the project and setup various integrations.

Instead of
```
mkdir WebKitBuild/Release
cd WebKitBuild/Release
cmake -DPORT=gtk -DDEVELOPER_MODE=ON -DCMAKE_BUILD_TYPE=Release ../..
cmake --build .
```
it's now possible to do just
```
cmake --preset=gtk-release
cmake --build --preset=gtk-release
```

* CMakePresets.json: Added.

Canonical link: https://commits.webkit.org/261422@main
